samples_project/Assets/SampleViewer/Samples/Measure/Measure.cs

```diff
@@ -0,0 +1,298 @@
+                    featurePoints.Add(lineMarker);
+                   
+                   
+                }
+                //add this point to stops and also to feature points where stop is user-drawed, and feature points is a collection of user-drawed and interpolated
+                stops.Push(lineMarker);
+               
+            }
+        }
+        RenderLine(ref featurePoints);
+        RebaseLine();
+
+    }
+    
+    private void Interpolate(GameObject start, GameObject end, List<GameObject> featurePoints)
+    {
+        ArcGISLocationComponent startLocation = start.GetComponent<ArcGISLocationComponent>();
+        ArcGISLocationComponent endLocation = end.GetComponent<ArcGISLocationComponent>();
+
+        ArcGISPoint startPoint = new ArcGISPoint(startLocation.Position.X, startLocation.Position.Y, startLocation.Position.Z, spatialRef);
+        ArcGISPoint endPoint = new ArcGISPoint(endLocation.Position.X, endLocation.Position.Y, endLocation.Position.Z, spatialRef);
+
+        double d = ArcGISGeometryEngine.DistanceGeodetic(startPoint, endPoint, new ArcGISLinearUnit((ArcGISLinearUnitId)9001), new ArcGISAngularUnit(unitDegree), ArcGISGeodeticCurveType.Geodesic).Distance;
+        float n = Mathf.Floor((float)d / InterpolationInterval);
+        double dx = (end.transform.position.x - start.transform.position.x) / n;
+        double dz = (end.transform.position.z - start.transform.position.z) / n;
+
+        prePoint = startPoint;
+        GameObject pre = start;
+
+        //calculate n-1 intepolation points/n-1 segments because the last segment is already created by the end point 
+        for (int i=0;i<n-1;i++)
+        {
+            GameObject next = Instantiate(InterpolationMarker, arcGISMapComponent.transform);
+
+            //calculate transform of next point
+            float nextX = pre.transform.position.x + (float)dx;
+            float nextZ = pre.transform.position.z + (float)dz;
+            next.transform.position = new Vector3(nextX, 0, nextZ);
+
+            //set default location component of next point
+            next.GetComponent<ArcGISLocationComponent>().enabled = true;
+            next.GetComponent<ArcGISLocationComponent>().Rotation = new ArcGISRotation(0, 90, 0);
+
+            //define height
+            SetElevation(next);
+ 
+            featurePoints.Add(next);
+
+            prePoint = nextPoint;
+            pre = next;
+        }
+
+    }
+    
+    private ArcGISPoint HitToGeoPosition(RaycastHit hit, float yOffset = 0)
+    {
+        var worldPosition = math.inverse(arcGISMapComponent.WorldMatrix).HomogeneousTransformPoint(hit.point.ToDouble3());
+
+        var geoPosition = arcGISMapComponent.View.WorldToGeographic(worldPosition);
+        var offsetPosition = new ArcGISPoint(geoPosition.X, geoPosition.Y, geoPosition.Z + yOffset, geoPosition.SpatialReference);
+
+        return GeoUtils.ProjectToSpatialReference(offsetPosition, spatialRef);
```

Comment:
```suggestion
        var offsetPosition = new ArcGISPoint(geoPosition.X, geoPosition.Y, geoPosition.Z + yOffset, geoPosition.SpatialReference);

        return GeoUtils.ProjectToSpatialReference(offsetPosition, spatialRef);
```

No reason to do the projection here.